api/src/main/java/org/pmiops/workbench/db/model/DbInstitutionEmailAddress.java

```diff
@@ -15,51 +15,58 @@
 @Table(name = "institution_email_address")
 public class DbInstitutionEmailAddress {
 
-  @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  @Column(name = "institution_email_address_id")
   private long institutionEmailAddressId;
-
-  @ManyToOne(fetch = FetchType.LAZY)
-  @JoinColumn(name = "institution_id")
   private DbInstitution institution;
-
-  @Column(name = "email_address", nullable = false)
   private String emailAddress;
 
   public DbInstitutionEmailAddress() {}
 
-  public DbInstitutionEmailAddress(DbInstitution institution, String emailAddress) {
-    this.institution = institution;
-    this.emailAddress = emailAddress;
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "institution_email_address_id", nullable = false)
+  public long getInstitutionEmailAddressId() {
+    return institutionEmailAddressId;
   }
 
+  public DbInstitutionEmailAddress setInstitutionEmailAddressId(long institutionEmailAddressId) {
+    this.institutionEmailAddressId = institutionEmailAddressId;
+    return this;
+  }
+
+  @ManyToOne(cascade = CascadeType.ALL)
+  @JoinColumn(name = "institution_id", nullable = false)
   public DbInstitution getInstitution() {
     return institution;
   }
 
+  public DbInstitutionEmailAddress setInstitution(DbInstitution institution) {
+    this.institution = institution;
+    return this;
+  }
+
+  @Column(name = "email_address", nullable = false)
   public String getEmailAddress() {
     return emailAddress;
   }
 
+  public DbInstitutionEmailAddress setEmailAddress(String emailAddress) {
+    this.emailAddress = emailAddress;
+    return this;
+  }
+
   @Override
   public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (!(o instanceof DbInstitutionEmailAddress)) {
-      return false;
-    }
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
 
     DbInstitutionEmailAddress that = (DbInstitutionEmailAddress) o;
 
-    return institutionEmailAddressId == that.institutionEmailAddressId
-        && institution.equals(that.institution)
-        && emailAddress.equals(that.emailAddress);
+    return Objects.equals(institution, that.institution)
+        && Objects.equals(emailAddress, that.emailAddress);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(institutionEmailAddressId, institution, emailAddress);
+    return Objects.hash(institution, emailAddress);
```

Comment:
So this class allows the same email address to belong to more than one institution?

Reply:
yes.  We have no requirement to exclude this.